File: exchanges/binance.py
import requests


def get_binance_spot_trade_link(currency1: str, currency2: str) -> str:
    return f"https://www.binance.com/ru/trade/{currency1}_{currency2}?type=spot"


def get_binance_futures_trade_link(currency1: str, currency2: str) -> str:
    return f"https://www.binance.com/ru/futures/{currency1}{currency2}"


def get_binance_funding_rates() -> dict[str, dict[str, float]]:
    funding_rates = requests.get("https://fapi.binance.com/fapi/v1/premiumIndex")

    return {
        funding_rate["symbol"]: {
            "funding_rate": float(funding_rate["lastFundingRate"]) * 100,
            "index_price": float(funding_rate["indexPrice"]),
            "mark_price": float(funding_rate["markPrice"]),
            "next_funding_time": funding_rate["nextFundingTime"],
            "predicted_funding_rate": "-",
        }
        for funding_rate in funding_rates.json()
    }

File: exchanges/binance.js
const axios = require('axios');

const { ExchangeName } = require('../constants');
const { formatFundingRate, getTimeString } = require('../utils');

class Binance {
  getSpotTradeLink(symbol) {
    const currency = symbol.split('USDT')[0];
    return `https://www.binance.com/ru/trade/${currency}_USDT?type=spot`;
  }

  getFuturesTradeLink(symbol) {
    return `https://www.binance.com/ru/futures/${symbol}`;
  }

  async getFundingRates() {
    try {
      const { data: fundingRates } = await axios.get(
        'https://fapi.binance.com/fapi/v1/premiumIndex'
      );
      const { data: fundingIntervals } = await axios.get(
        'https://fapi.binance.com/fapi/v1/fundingInfo'
      );

      return fundingRates.reduce((acc, fundingRate) => {
        const symbol = fundingRate.symbol;
        const fundingInterval =
          fundingIntervals.find((fundingInterval) => fundingInterval.symbol === symbol)
            ?.fundingIntervalHours ?? 8;

        return {
          ...acc,
          [symbol.replace(/^10+/g, '')]: {
            fundingRate: formatFundingRate(fundingRate.lastFundingRate),
            indexPrice: fundingRate.indexPrice,
            markPrice: fundingRate.markPrice,
            nextFundingTime: getTimeString(fundingRate.nextFundingTime),
            fundingInterval,
            predictedFundingRate: '-',
            spotLink: this.getSpotTradeLink(symbol.replace(/^10+/g, '')),
            futuresLink: this.getFuturesTradeLink(symbol),
            multiplier: symbol.match(/^10+/g)?.[0] ?? 1,
          },
        };
      });
    } catch (err) {
      console.log(`Ошибка получения данных фандинга ${ExchangeName.BINANCE}. ${err?.message}`);
    }
  }
}

module.exports = new Binance();
